ui-testing

naumvd95 opened the pull request:
proofs = http://jenkins-tpi.bud.mirantis.net:8080/view/ui/job/10.0.ui.test_bonds/17/console

naumvd95 commented on servers/partner-ci/builders/common.prepare.sh:
```diff
@@ -20,15 +20,16 @@ fi
+if [[ "${UPDATE_MASTER}" == "true" ]] && [[ ${FUEL_RELEASE} != *"80"* ]] && [[ ${FUEL_RELEASE} != *"100"* ]]; then
   [ ${SNAPSHOTS_ID} ] && export SNAPSHOTS_ID=${SNAPSHOTS_ID} || export SNAPSHOTS_ID=${CUSTOM_VERSION:10}
+elif [[ ${FUEL_RELEASE} == *"100"* ]]; then
+  export SNAPSHOTS_ID="from_iso"
```
pay attention!!
there are some troubles with it
need to refactor testing-code
**2017-02-13 14:29:55 - INFO environment.py:237 -- Try to find environment '10.0.ui.released''**

rkhozinov replied:
**10.0.ui.released** --> **10.0.ui.<iso_number>**

naumvd95 replied:
done

rkhozinov reviewed (changes requested):
Separate download job to another pull-request

rkhozinov commented on servers/partner-ci/builders/common.prepare.sh:
```diff
@@ -20,15 +20,16 @@ fi
 
 
-if [[ $ISO_FILE == *"Mirantis"* ]]; then
+if [[ $ISO_FILE == *"Mirantis"* ]] || [[ "${FUEL_RELEASE_VERSION}" == "10" ]] ; then
```
join conditions

naumvd95 replied:
there is different conditions
need to refactor this cases
i will make the issue

naumvd95 replied:
https://github.com/rkhozinov/jenkins-jobs/issues/175

rkhozinov replied:
`FUEL_RELEASE_VERSION` - where do we initialize the var?

naumvd95 replied:
common.yaml:            FUEL_RELEASE_VERSION={fuel_release_checker};

rkhozinov commented on servers/partner-ci/common.yaml:
```diff
@@ -194,6 +194,10 @@
           name: SNAPSHOTS_ID
           default: '{snapshots_id}'
           description: 'snapshots number, "lastSuccessfulBuild" or "released"'
+      - string:
```
add to common props

naumvd95 replied:
done

rkhozinov commented on servers/partner-ci/common.yaml:
```diff
@@ -630,7 +634,7 @@
       - shell: |
           #!/bin/bash -xeu
-          UBUNTU_MIRROR_URL="${MIRROR_HOST:?}${UBUNTU_MIRROR_ID:?}/"
+          UBUNTU_MIRROR_URL="${MIRROR_HOST:?}/pkgs/snapshots/${UBUNTU_MIRROR_ID:?}/"
```
add to the common props the next property: 
**MIRROR_HOST=http://mirror.seed-cz1.fuel-infra.org/pkgs/snapshots/**

naumvd95 replied:
done

rkhozinov commented on servers/partner-ci/common.yaml:
```diff
@@ -641,7 +645,7 @@
           fi
 
-          echo "MIRROR_UBUNTU="${UBUNTU_REPOS:?}" >> specific.properties
+          echo MIRROR_UBUNTU="${UBUNTU_REPOS:?}" >> specific.properties
```
"MIRROR_UBUNTU=${UBUNTU_REPOS:?}"

rkhozinov replied:
java properties: 

MIRROR=asdfadsf

bash properties
MIRROR="asdf"

rkhozinov replied:
`echo "MIRROR_UBUNTU=${UBUNTU_REPOS:?}"`

naumvd95 replied:
done

rkhozinov commented on servers/partner-ci/builders/ui.test.sh:
```diff
@@ -31,7 +31,7 @@ sh -ex "utils/jenkins/system_tests.sh"  \
    -t test                              \
    -o --group="${FUEL_QA_TEST_GROUP:?}" \
-   -i ${ISO_FILE:?}
+   -i ${ISO_STORAGE}/${ISO_FILE:?}
```
`-i ${ISO_STORAGE:?}/${ISO_FILE:?}`

naumvd95 replied:
done

rkhozinov commented on servers/partner-ci/builders/common.prepare.sh:
```diff
@@ -45,21 +46,19 @@ if [[ $ISO_FILE == *"Mirantis"* ]]; then
 export ENV_NAME="${ENV_PREFIX}.${SNAPSHOTS_ID}"
 export VENV_PATH=${VENV_PATH:-"$HOME/$FUEL_RELEASE-venv"}
+
+[[ -d $VENV_PATH ]] || virtualenv --clear "${VENV_PATH}"
```
`[[ -d $VENV_PATH ]] `--> `[ -d $VENV_PATH ] `OR `test -d $VENV_PATH`

naumvd95 replied:
done

Commit:
refactor downloading-iso job

Commit:
refactoring prepare job-file, fix ui-job bugs

rkhozinov commented on servers/partner-ci/builders/build-job-download-iso.sh:
```diff
@@ -8,20 +8,20 @@ FUEL_RELEASE=$(echo $VERSION | tr -d . )
 echo "Description string: $NODE_NAME $VERSION"
 
+ISO_STORAGE="/storage/downloads"
 aria2c --seed-time 0 "${MAGNET_LINK}" -d "${WORKSPACE}"
```
why you don't download iso directly to `/storage/downloads`?

naumvd95 replied:
we already talk about that
before moving we need to define ISO_FILE=$(ls *.iso)

rkhozinov replied:
aria2c provides information about downloaded files

rkhozinov commented on servers/partner-ci/builders/common.prepare.sh:
```diff
@@ -45,21 +48,19 @@ if [[ $ISO_FILE == *"Mirantis"* ]]; then
 ## the fuel-qa branch is determined by a fuel-iso name.
 
 case "${FUEL_RELEASE}" in
-  *10* ) export REQS_BRANCH="stable/newton" ;;
```
join with line 30

naumvd95 replied:
done

rkhozinov commented on servers/partner-ci/builders/ui.test.sh:
```diff
@@ -6,7 +6,7 @@ fuel_release=$(echo $ISO_FILE | cut -d- -f2 | tr -d '.iso')
 
 if [ "${SNAPSHOTS_ID}" != "released" ]; then
-  if [[ "${UPDATE_MASTER}" == "true" ]] && [[ ${FUEL_RELEASE} != *"80"* ]]; then
+  if [[ "${UPDATE_MASTER}" == "true" ]] && [[ ${FUEL_RELEASE} != *"80"* ]] && [[ ${FUEL_RELEASE} != *"100"* ]]; then
```
try to use switch or regex groups

naumvd95 replied:
done

rkhozinov commented on servers/partner-ci/common.yaml:
```diff
@@ -119,16 +115,17 @@
             FORCE_DISABLE_UPDATES=false;
             FUEL_STATS_ENABLED=false;
             GIT_CURL_VERBOSE=false;
+            ISO_STORAGE=/storage/downloads;
```
ok

rkhozinov commented on servers/partner-ci/common.yaml:
```diff
@@ -119,16 +115,17 @@
             SLAVE_NODE_MEMORY=4096;
             SNAPSHOT_ARTIFACTS_FILE=snapshots.params;
             SNAPSHOT_OUTPUT_FILE=extra_repos.sh;
+            VENV_PATH=$HOME/90-venv;
```
90-venv is used for fuel90
100-venv is used for fuel100

VENV_PATH wasn't a constant

naumvd95 replied:
this is common variable, seems this is right
in spec-props i'll define VENV_PATH=$HOME/100-venv;

rkhozinov commented on servers/partner-ci/common.yaml:
```diff
@@ -641,7 +639,7 @@
           fi
 
-          echo "MIRROR_UBUNTU="${UBUNTU_REPOS:?}" >> specific.properties
+          echo "MIRROR_UBUNTU=${UBUNTU_REPOS:?}" >> specific.properties
```
ok

rkhozinov commented on servers/partner-ci/common.yaml:
```diff
@@ -210,10 +207,11 @@
             FUEL_RELEASE_VERSION={fuel_release_checker};
             FUEL_STATS_ENABLED=false;
             GIT_CURL_VERBOSE=false;
+            ISO_STORAGE=/storage/downloads;
```
ok

rkhozinov commented on servers/partner-ci/common.yaml:
```diff
@@ -210,10 +207,11 @@
             KVM_USE={kvm_use};
             MAKE_SNAPSHOT=true;
-            MIRROR_HOST=mirror.seed-cz1.fuel-infra.org;
+            MIRROR_HOST=http://mirror.seed-cz1.fuel-infra.org/pkgs/snapshots/;
```
ok

rkhozinov commented on servers/partner-ci/builders/common.prepare.sh:
```diff
@@ -20,22 +20,25 @@ fi
+  esac
 fi
 
 [ -z "${SNAPSHOTS_ID}" ] && { echo SNAPSHOTS_ID is empty; exit 1; }
```
remove the check `[ -z "${SNAPSHOTS_ID}" ] && { echo SNAPSHOTS_ID is empty; exit 1; }`

rkhozinov commented on servers/partner-ci/builders/common.prepare.sh:
```diff
@@ -20,22 +20,25 @@ fi
 [ -z $ISO_FILE ] && (echo "ISO_FILE variable is empty"; exit 1)
```
remove `[ -z $ISO_FILE ] && (echo "ISO_FILE variable is empty"; exit 1)`
and
change 
`if [[ ${ISO_FILE:?} == *"Mirantis"* ]] || [[ "${FUEL_RELEASE_VERSION}" == "10" ]] `

naumvd95 replied:
done

Commit:
code-style fixes

naumvd95 committed:
refactoring common-prepare code

naumvd95 committed:
code-style fixex, description added

naumvd95 committed:
fix to avoid null-overwrite

Commit:
common: huge common refactoring

Commit:
common: huge common refactoring

Commit:
ui: 10 fuel ui rf related changes

naumvd95 committed:
Merge branch 'common.prepare.rf' of github.com:rkhozinov/jenkins-jobs into common.prepare.rf

Conflicts:
	servers/partner-ci/common.yaml

naumvd95 committed:
some bugfix

code-style fix

rkhozinov committed:
Merge pull request #180 from rkhozinov/common.prepare.rf-mk1

code-style fix for huge common rf

rkhozinov committed:
Merge pull request #179 from rkhozinov/common.prepare.rf

common: huge common refactoring

rkhozinov commented on servers/partner-ci/builders/common.prepare.sh:
```diff
@@ -165,38 +105,15 @@ function dospy_list {
   get_venv_requirements
-  [ -d $VENV_PATH ] && prepare_venv || { echo "$VENV_PATH doesn't exist $VENV_PATH will be recreated"; recreate_venv; }
+  [ -d $VENV_PATH ] && prepare_venv || \
+    { echo "$VENV_PATH doesn't exist $VENV_PATH will be recreated"; recreate_venv; }
```
the statement's logic confuses

naumvd95 committed:
global merging

naumvd95 committed:
fix to avoid travis checks

naumvd95 committed:
fix incorrect logic

all necessary methods are already in "recreate_venv" function

rkhozinov commented on servers/partner-ci/9.0/contrail.yaml:
```diff
@@ -392,11 +392,12 @@
+      - 20-get-props
+      - 21-get-baremetal-props
+      - 30-get-additional-updates
+      - 40-get-snapshots
```
join additional updates with get-snapshots

naumvd95 committed:
builders/:refactor tests.sh for jobs to associate with new macroses in common

common: join 2 macroses in 1

rkhozinov commented on servers/partner-ci/common.yaml:
```diff
@@ -589,6 +566,23 @@
             - shell: |
                 #!/bin/bash -xeu
+                if [[ "${UPDATE_MASTER}" == "true" ]]; then
+                cat << SPECIFIC_PROPERTIES > specific.properties
```
indent

naumvd95 committed:
common: change injecting-case

common: previously-specific properties now named as macros-builder name
common: remove ";" delimeter from pure injecting case
common: inject-case bugfix

rkhozinov commented on servers/partner-ci/common.yaml:
```diff
@@ -598,16 +587,22 @@
                 python "${CONV_SNAP_PATH:?}"
 
+                . $SNAPSHOT_OUTPUT_FILE
+                echo "EXTRA_RPM_REPOS=${EXTRA_RPM_REPOS}" >> get-snapshots.properties
```
optimize multiple ehos with cat

naumvd95 committed:
common: refactor file-writing cases

naumvd95 committed:
common:remove useless defining var
